src/identity-providers/ManageOrderDialog.tsx

```diff
@@ -34,28 +45,51 @@ export const ManageOrderDialog = ({
   const adminClient = useAdminClient();
   const { addAlert, addError } = useAlerts();
 
-  const [alias, setAlias] = useState("");
   const [liveText, setLiveText] = useState("");
   const [order, setOrder] = useState(
-    providers.map((provider) => provider.alias!)
+    sortBy(providers, ["config.guiOrder", "alias"]).map(
+      (provider) => provider.alias!
+    )
   );
 
-  const onDragStart = (id: string) => {
-    setAlias(id);
-    setLiveText(t("common:onDragStart", { item: id }));
+  const onDrag = (source: DraggableItemPosition) => {
+    setLiveText(
+      t("common:onDragStart", { item: providers[source.index].alias })
+    );
+    // Return true to allow drag
+    return true;
   };
 
-  const onDragMove = () => {
-    setLiveText(t("common:onDragMove", { item: alias }));
+  const onDragMove = (
+    source: DraggableItemPosition,
+    dest?: DraggableItemPosition
+  ) => {
+    const newText = dest
+      ? t("common:onDragMove", { item: providers[source.index].alias })
+      : t("common:onDragInvalid");
+    if (newText !== liveText) {
+      setLiveText(newText);
+    }
   };
 
-  const onDragCancel = () => {
-    setLiveText(t("common:onDragCancel"));
-  };
+  const onDrop = (
+    source: DraggableItemPosition,
+    dest?: DraggableItemPosition
+  ): boolean => {
+    if (dest) {
+      const result = [...order];
+      const [removed] = result.splice(source.index, 1);
+      result.splice(dest.index, 0, removed);
+      setOrder(result);
 
-  const onDragFinish = (providerOrder: string[]) => {
-    setLiveText(t("common:onDragFinish", { list: providerOrder }));
-    setOrder(providerOrder);
+      setLiveText(
+        t("common:onDragFinish", { list: providers.map((p) => p.alias) })
```

Comment:
You could get rid of this one char variable with some destructuring.
```suggestion
        t("common:onDragFinish", { list: providers.map(({ alias }) => alias) })
```